include/small/ibuf.h

```diff
@@ -124,23 +126,27 @@ ibuf_reserve_slow(struct ibuf *ibuf, size_t size);
 static inline void *
 ibuf_reserve(struct ibuf *ibuf, size_t size)
 {
-	if (ibuf->wpos + size <= ibuf->end)
+	if (ibuf->wpos + size <= ibuf->end) {
+		ASAN_UNPOISON_MEMORY_REGION(ibuf->wpos, ibuf->end - ibuf->wpos);
 		return ibuf->wpos;
+	}
 	return ibuf_reserve_slow(ibuf, size);
 }
 
 static inline void *
 ibuf_alloc(struct ibuf *ibuf, size_t size)
 {
 	void *ptr;
-	if (ibuf->wpos + size <= ibuf->end)
+	if (ibuf->wpos + size <= ibuf->end) {
+		ASAN_UNPOISON_MEMORY_REGION(ibuf->wpos, ibuf->end - ibuf->wpos);
```

Comment:
Added a comment here and in the buffer.lua of Tarantool patch.
```c
@@ -146,6 +146,11 @@ ibuf_alloc(struct ibuf *ibuf, size_t size)
 {
        void *ptr;
        if (ibuf->wpos + size <= ibuf->end) {
+               /*
+                * In case of using same buffer we need to unpoison newly
+                * allocated memory after previous ibuf_alloc or poison after
+                * newly allocated memory after after previous ibuf_reserve.
+                */
                ASAN_UNPOISON_MEMORY_REGION(ibuf->wpos, ibuf->end - ibuf->wpos);
                ptr = ibuf->wpos;
        } else {
```